e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,318 @@
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Adding deployment")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Check that the deployment was created")
+	if err := nt.Validate("hello-world", "dir", &appsv1.Deployment{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Check that specified field name is present")
+	if err := nt.Validate("dir", "", &v1.NamespaceConfig{}, nomostest.NamspaceConfigHasMetadataName("hello-world")); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Check that unspecified field creationTimestamp is not present")
+	if err := nt.Validate("dir", "", &v1.NamespaceConfig{}, nomostest.NamspaceConfigHasMetadataCreationTimestamp(metav1.Time{})); err != nil {
+		nt.T.Fatal("NamespaceConfig has default field creationTimestamp which was not specified")
+	}
+}
+
+func TestRecreateSyncDeletedByUser(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation2, ntopts.SkipMultiRepo)
+
+	nt.T.Log("Add a deployment to a directory")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/dir-namespace.yaml", yamlDir), "acme/namespaces/dir/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/deployment-helloworld.yaml", yamlDir), "acme/namespaces/dir/deployment.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add a deployment to a directory")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Ensure that the system created a sync for the deployment")
+	if err := nt.Validate("deployment.apps", "default", &v1.Sync{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Force-delete the sync from the cluster")
+	sync := &v1.Sync{}
+	if err := nt.Get("deployment.apps", "default", sync); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.Delete(sync); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Ensure that the system re-created the force-deleted sync")
+	_, err := nomostest.Retry(60*time.Second, func() error {
+		err := nt.Validate("deployment.apps", "default", &v1.Sync{})
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func TestDeletingClusterConfigRecoverable(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation2, ntopts.SkipMultiRepo)
+
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/dir-namespace.yaml", yamlDir), "acme/namespaces/dir/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add dir namespace")
+	nt.WaitForRepoSyncs()
+
+	if err := nt.Validate("dir", "", &corev1.Namespace{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Forcefully delete clusterconfigs and verify recovery")
+	cmClusterConfig := &v1.ClusterConfig{}
+
+	if err := nt.Get("config-management-cluster-config", "", cmClusterConfig); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.Delete(cmClusterConfig); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	cmCRDCLusterConfig := &v1.ClusterConfig{}
+	if err := nt.Get("config-management-crd-cluster-config", "", cmCRDCLusterConfig); err != nil {
```

Comment:
I don't see this resource in the equivalent bats test. What changed?

Reply:
i was trying to simulate `kubectl delete clusterconfig --all`, but seems like this is not a good way to do it. I've changed it to use `nt.DeleteAllOf`